### PR TITLE
Remove persist_invalid from order

### PR DESCRIPTION
**Pull request by cbrunsdon**
This plugs up our "leaky database". You'll also notice no test removals.

Also no lines of code added at all, for that matter.

### CODE REVIEW

**Comment by DangerDawson** [comment]
I have also been seeing a stack level to deep when running the payment steps due to this code after upgrading from Spree 2.2.2 to the latest version of 2.2-stable.

This is due to that once a spec has finished the after rollback is being triggered, and trying to save the payment although the order no longer exists ( it has been rolled back in the transaction ), which causes an exception which then triggers another rollback etc. 

Another alternative is to stub out the rollback in the spec helper and only explicitly enable it for specs that are testing it

**Comment by cbrunsdon** [comment]
@DangerDawson are you sure you're talking about this PR? It hasn't been merged yet, and I'm not sure its runnable against 2-2-stable.

File: core/spec/models/spree/stock/quantifier_spec.rb [hunk 1]
```diff
@@ -12,8 +12,6 @@ module Spree
   module Stock
     describe Quantifier, :type => :model do
 
-      before(:all) { Spree::StockLocation.destroy_all } #FIXME leaky database
```
Comment (JDutil) [inline core/spec/models/spree/stock/quantifier_spec.rb]: For some reason I'm having to add this one back to get specs passing on master, but they passed fine on 2-4 without this.

Comment (gmacdougall) [inline core/spec/models/spree/stock/quantifier_spec.rb]: Odd. The database might still be leaking somewhere...

**Comment by huoxito** [comment]
was investigating infinite loop issues on rails 4.2 branch, wish I had seen this before lol poor me

**Pull request closed without merging**